openmc/mgxs_library.py

```diff
@@ -1634,6 +1685,267 @@ def set_inverse_velocity_mgxs(self, inverse_velocity, temperature=294.,
+            # Nothing to do as the same structure was requested
+            return xsdata
+
+        xsdata.representation = target_representation
+        # We have different actions depending on the representation conversion
+        if target_representation == 'isotropic':
+            # This is not needed for the correct functionality, but these
+            # values are changed back to None for clarity
+            xsdata._num_polar = None
+            xsdata._num_azimuthal = None
+        elif target_representation == 'angle':
+            xsdata.num_polar = num_polar
+            xsdata.num_azimuthal = num_azimuthal
+
+        # Reset xs_shapes so it is recalculated the next time it is needed
+        xsdata._xs_shapes = None
+
+        for i, temp in enumerate(xsdata.temperatures):
+            for xs in ['total', 'absorption', 'fission', 'nu_fission',
+                       'scatter_matrix', 'multiplicity_matrix',
+                       'prompt_nu_fission', 'delayed_nu_fission',
+                       'kappa_fission', 'chi', 'chi_prompt', 'chi_delayed',
+                       'beta', 'decay_rate', 'inverse_velocity']:
+                # Get the original data
+                orig_data = getattr(self, '_' + xs)[i]
+                if orig_data is not None:
+                    if target_representation == 'isotropic':
+                        # Since we are going from angle to isotropic, the
+                        # current data is just the average over the angle bins
+                        new_data = orig_data.mean(axis=(0, 1))
+                    elif target_representation == 'angle':
+                        # Since we are going from isotropic to angle, the
+                        # current data is just copied for every angle bin
+                        new_shape = (num_polar, num_azimuthal) + \
+                            orig_data.shape
+                        new_data = np.resize(orig_data, new_shape)
+                    setter = getattr(xsdata, 'set_' + xs)
+                    setter(new_data, temp)
+
+        return xsdata
+
+    def convert_scatter_format(self, target_format, target_order=None):
+        """Produce a new MGXSLibrary object with the same data, but converted
+        to the new scatter format and order
+
+        Parameters
+        ----------
+        target_format : {'tabular', 'legendre', 'histogram'}
+            Representation of the scattering angle distribution
+        target_order : int
+            Either the Legendre target_order, number of bins, or number of
+            points used to describe the angular distribution associated with
+            each group-to-group transfer probability
+
+        Returns
+        -------
+        openmc.XSdata
+            Multi-group cross section data with the same data as in self, but
+            represented as specified in :param:`target_format`.
+
+        """
+
+        from scipy.interpolate import interp1d
```

Comment:
I agree they can be separated but I'm not sure the benefit is worth the effort since everything is pretty well segregated in to its own if/else block.